apps/api/src/app/workflows-v2/util/template-parser/liquid-parser.ts

```diff
@@ -245,6 +246,27 @@ function parseByLiquid({
         const isAllowedVariable = isPropertyAllowed(variableSchema, variableName);
         if (isAllowedVariable) {
           validVariables.push({ name: variableName, output: rawOutput });
+          if (filters.length > 0) {
+            filters.forEach((filter) => {
+              const { args } = filter;
+              const firstArg = args[0];
+              if (
+                filter.name === 'toSentence' &&
+                args.length > 0 &&
+                'content' in firstArg &&
+                typeof firstArg.content === 'string' &&
+                firstArg.content.startsWith('payload.')
+              ) {
```

Comment:
also only when the variable name matches the `steps.x.events`, otherwise we should not treat it as the variable